src/assembly/local/quadrature_table.rs

```diff
@@ -147,6 +147,36 @@ where
             data: self.data,
         }
     }
+
+    /// Replaces the data of the quadrature table with the given data.
+    pub fn with_data<NewData>(self, data: NestedVec<NewData>) -> GeneralQuadratureTable<T, GeometryDim, NewData> {
+        GeneralQuadratureTable {
+            points: self.points,
+            weights: self.weights,
+            data: data,
+        }
+    }
+
+    pub fn with_data_from_fn<NewData>(
+        self,
+        mut data_fn: impl FnMut(usize, &OPoint<T, GeometryDim>) -> NewData,
+    ) -> GeneralQuadratureTable<T, GeometryDim, NewData> {
+        let mut data = NestedVec::new();
+
+        for (element_index, points) in self.points.iter().enumerate() {
+            let mut arr = data.begin_array();
+
+            for point in points {
+                arr.push_single(data_fn(element_index, point));
+            }
+        }
+
+        GeneralQuadratureTable {
+            points: self.points,
+            weights: self.weights,
+            data: data,
+        }
+    }
```

Comment:
This appears to be unrelated to MSH, and seems to be the same functionality as in #49 ? Also applies to code below.